yb-voyager/src/migassessment/gather.go

```diff
@@ -0,0 +1,566 @@
+	}
+
+	for _, path := range possiblePathsForScript {
+		if utils.FileOrFolderExists(path) {
+			log.Infof("found the gather assessment metadata script at: %s", path)
+			return path, nil
+		}
+	}
+
+	return "", fmt.Errorf("script not found in possible paths: %v", possiblePathsForScript)
+}
+
+func runGatherAssessmentMetadataScript(scriptPath string, envVars []string, workingDir string, scriptArgs ...string) error {
+	cmd := exec.Command(scriptPath, scriptArgs...)
+	log.Infof("running script: %s", cmd.String())
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, envVars...)
+	cmd.Dir = workingDir
+	cmd.Stdin = os.Stdin
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("error creating stdout pipe: %w", err)
+	}
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("error creating stderr pipe: %w", err)
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("error starting gather assessment metadata script: %w", err)
+	}
+
+	go func() {
+		scanner := bufio.NewScanner(stderr)
+		for scanner.Scan() {
+			log.Errorf("[stderr of script]: %s", scanner.Text())
+			fmt.Printf("%s\n", scanner.Text())
+		}
+	}()
+
+	scanner := bufio.NewScanner(stdout)
+	for scanner.Scan() {
+		log.Infof("[stdout of script]: %s", scanner.Text())
+		fmt.Printf("%s\n", scanner.Text())
+	}
+
+	err = cmd.Wait()
+	if err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				if status.ExitStatus() == 2 {
+					log.Infof("Exit without error as user opted not to continue in the script.")
+					os.Exit(0)
+				}
+			}
+		}
+		return fmt.Errorf("error waiting for gather assessment metadata script to complete: %w", err)
+	}
+	return nil
+}
+
+// runGatherAssessmentMetadataScriptBuffered runs the metadata collection script with buffered output
+// and sends progress updates to the provided channel. This version is safe for parallel execution.
+func runGatherAssessmentMetadataScriptBuffered(
+	scriptPath string,
+	envVars []string,
+	nodeName string,
+	progressChan chan<- NodeProgress,
+	workingDir string,
+	scriptArgs ...string,
+) error {
+	cmd := exec.Command(scriptPath, scriptArgs...)
+	log.Infof("[%s] running script: %s", nodeName, cmd.String())
+	cmd.Env = os.Environ()
+	cmd.Env = append(cmd.Env, envVars...)
+	cmd.Dir = workingDir
+	// Don't set stdin for parallel execution to avoid conflicts.
+	// Not needed anyway since we pass skip_checks=true (no prompts).
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return fmt.Errorf("error creating stdout pipe: %w", err)
+	}
+
+	stderr, err := cmd.StderrPipe()
+	if err != nil {
+		return fmt.Errorf("error creating stderr pipe: %w", err)
+	}
+
+	err = cmd.Start()
+	if err != nil {
+		return fmt.Errorf("error starting gather assessment metadata script: %w", err)
+	}
+
+	// Report starting status
+	if progressChan != nil {
```

Comment:
we seem to be checking `if progressChan != nil` multiple times in this function. 
Can we instead 
- check if once at the beginning. if it's nil, return error. 
- thereafter, we can safely just append onto that channel.